core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1556,6 +1557,11 @@ class ReplicaManager(val config: KafkaConfig,
             error(s"Error while making broker the follower for partition $partition with leader " +
               s"$newLeaderBrokerId in dir $dirOpt", e)
             responseMap.put(partition.topicPartition, Errors.KAFKA_STORAGE_ERROR)
+          case e: ZooKeeperClientException =>
```

Comment:
It's probably better to do this in Partition.makeFollower() instead of here. That way, we only skip partitions that have incurred ZK error.

Also, the same ZK exception can happen in Partition.makeLeader(). So, we want to do the same thing there as well.

Reply:
👍

Reply:
@junrao -- let me know if there's additional changes to be made. Thanks.